### PR TITLE
caller provided serializer storage

### PR DESCRIPTION
**Pull request by vinniefalco**

### CODE REVIEW

**Comment by vinniefalco** [comment]
Already done https://github.com/boostorg/json/pull/832

File: include/boost/json/detail/stack.hpp [hunk 1]
```diff
@@ -21,79 +21,72 @@ class stack
 {
     storage_ptr sp_;
     std::size_t cap_ = 0;
-    std::size_t size_ = 0;
-    char* buf_ = nullptr;
+    std::size_t size0_ = 0; // lo stack
+    std::size_t size1_ = 0; // hi stack
```
Comment (grisumbras) [inline include/boost/json/detail/stack.hpp:25]: I don't see where `size1_` is used

File: include/boost/json/detail/impl/stack.hpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    |<------>|<------>|<------>|
+      size0_            size1_
+*/
+
+template<class T>
+void
+stack::
+push(T const& t)
+{
+    if(size0_ + sizeof(T) >
```
[the remark this answers is not in the document]
Comment (vinniefalco) [reply include/boost/json/detail/impl/stack.hpp:32]: just easier to read I guess, no having to look elsewhere. but I see your point.

File: include/boost/json/impl/serialize.ipp [hunk 3]
```diff
@@ -109,8 +125,12 @@ std::string
 serialize(
     string_view sv)
 {
```
[the remark this answers is not in the document]
Comment (vinniefalco) [reply include/boost/json/impl/serialize.ipp:127]: That, I do not know.

**Pull request closed without merging**